cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/SourceControlManagementHttpHandler.java

```diff
@@ -0,0 +1,111 @@
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.HttpResponseStatus;
+
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+
+/**
+ * {@link io.cdap.http.HttpHandler} for source control management.
+ */
+@Path(Constants.Gateway.API_VERSION_3 + "/namespaces/{namespace-id}/repository")
+public class SourceControlManagementHttpHandler extends AbstractAppFabricHttpHandler {
+  private final SourceControlManagementService sourceControlService;
+  private static final Gson GSON = new Gson();
+
+  @Inject
+  SourceControlManagementHttpHandler(SourceControlManagementService sourceControlService) {
+    this.sourceControlService = sourceControlService;
+  }
+
+  @PUT
+  @Path("/")
+  @AuditPolicy(AuditDetail.REQUEST_BODY)
+  public void setRepository(FullHttpRequest request, HttpResponder responder,
+                            @PathParam("namespace-id") String namespaceId) throws Exception {
+    RepositoryConfigRequest repoRequest = getRepository(request);
+    if (!repoRequest.shouldValidate()) {
+      if (!repoRequest.isValid()) {
+        throw new BadRequestException(String.format("Invalid repository configuration: %s.", repoRequest));
+      }
+      sourceControlService.setRepository(validateNamespaceId(namespaceId), repoRequest);
+      responder.sendString(HttpResponseStatus.OK, String.format("Updated repository configuration for namespace '%s'.",
+                                                                namespaceId));
+    }
+
+    // TODO: CDAP-20252 add the validate logic once the SourceControlManager module is ready
+  }
+
+  @GET
+  @Path("/")
+  public void getRepository(FullHttpRequest request, HttpResponder responder,
+                            @PathParam("namespace-id") String namespaceId) throws Exception {
+    NamespaceId namespace = validateNamespaceId(namespaceId);
+    RepositoryConfig repository = sourceControlService.getRepository(namespace);
+
+    if (repository == null) {
+      throw new RepositoryNotFoundException(namespace);
+    }
+
+    responder.sendJson(HttpResponseStatus.OK, GSON.toJson(repository));
+  }
+
+  @DELETE
+  @Path("/")
+  public void deleteRepository(FullHttpRequest request, HttpResponder responder,
+                               @PathParam("namespace-id") String namespaceId) throws Exception {
+    sourceControlService.deleteRepository(validateNamespaceId(namespaceId));
+    responder.sendString(HttpResponseStatus.OK, String.format("Deleted repository configuration for namespace '%s'.",
+                                                              namespaceId));
+  }
+
+  private NamespaceId validateNamespaceId(String namespaceId) throws BadRequestException {
+    try {
+      return new NamespaceId(namespaceId);
+    } catch (IllegalArgumentException e) {
+      throw new BadRequestException("Namespace id can contain only alphanumeric characters or '_'.");
```

Comment:
use the exception's message instead of assuming what is wrong. `throw new BadRequestException(e.getMessage(), e);`